packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -363,12 +395,35 @@ const VirtualListBaseFactory = (type) => {
 		getMoreInfo = () => this.moreInfo
 
 		getGridPosition (index) {
+			if (this.props.type === 'NewVirtualList') {
+				const
+					{dimensionToExtent, primary, secondary, variableGridPositions} = this,
+					extent = Math.floor(index / dimensionToExtent),
+					firstIndexInExtent = extent * dimensionToExtent,
+					primaryPosition = variableGridPositions[firstIndexInExtent] ? variableGridPositions[firstIndexInExtent].position : extent * primary.gridSize,
+					secondaryPosition = (index % dimensionToExtent) * secondary.gridSize;
+
+				return {primaryPosition, secondaryPosition};
+			} else {
+				const {dimensionToExtent, primary, secondary} = this,
+					primaryPosition = Math.floor(index / dimensionToExtent) * primary.gridSize,
+					secondaryPosition = (index % dimensionToExtent) * secondary.gridSize;
+
+				return {primaryPosition, secondaryPosition};
+			}
+		}
+
+		// For NewVirtualList
+		getVariableGridPosition = (index) => {
```

Comment:
Note: this function assumes that a list always scrolls from the starting point 0.

Reply:
The name of this method indicates that it will return _**the starting position**_ corresponding to the input parameter `index`, but this returns _**the end position**_ in fact. It looks better to rename this method or to change the behavior.

Reply:
Good point! I will rename `getVariableGridPosition` as `getVariableGridBottomPosition`.